### DISCUSSION TITLE
Add constant for parkfile versions in ParkFile.cpp

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#19774] [Feature requests]
Currently all the parkfile patches use magic numbers. They could use constants instead like my speedy block brakes PR.

Thoughts?

**Comment by ZehMatt** [comment]
What do you mean? Do you mean the version number?

**Reply by spacek531** [reply]
Some examples:

https://github.com/OpenRCT2/OpenRCT2/blob/893ade7cccfe873c7d29dbbd715be1e2ece094d4/src/openrct2/park/ParkFile.cpp#L161

https://github.com/OpenRCT2/OpenRCT2/blob/893ade7cccfe873c7d29dbbd715be1e2ece094d4/src/openrct2/park/ParkFile.cpp#L1327

https://github.com/OpenRCT2/OpenRCT2/blob/893ade7cccfe873c7d29dbbd715be1e2ece094d4/src/openrct2/park/ParkFile.cpp#L330

The proposal is to add `constexpr const uint32_t` constants at the top of ParkFile.cpp.

**Reply by Broxzier** [reply]
While magic numbers are generally bad to have in the code, for version _numbers_ it makes perfect sense. You can see their value right there, and the body explains what it's for. I'd rather see code like this:

```cpp
if (version <= 12) { UpdateHybridAndSingleRailIds(); }
if (version <= 14) { UpdateAlphineId(); }
```
than:
```cpp
if (version <= ParkVersionChange::HybridAndSingleRailIdChange) { UpdateHybridAndSingleRailIds(); }
if (version <= ParkVersionChange::AlphineIdChange) { UpdateAlphineId(); }
```
(just making these names up btw). The order is also clear, and I don't see a benefit to naming these version changes, and these specific version checks are very unlikely to be used anywhere else in the code.

**Reply by ZehMatt** [reply]
Have to agree with @Broxzier, version numbers are typically nameless.
